R_program/Main.R

```diff
@@ -553,6 +566,181 @@ remove(temp)
 
 ############################# End MARKOV Weather Generator Code ##############################
 
+############################# Vegetation Code ##############################
+# only rescale space if requested.
+if(rescale_space){
+  # This code determines plant functional type relative abundance/composition
+  # and then scales STEPWAT2 parameters accordingly
+  source(vegetation.file)
+
+  # Array of plant functional type relative abundance/composition
+  relVegAbund <- estimate_STEPWAT_relativeVegAbundance(sw_weatherList)
+
+  # vectors that map rgroup names to the columns names of relVegAbund
+  Shrubs <- c("sagebrush", "shrub")
+  Forbs <- c("a.cool.forb","a.warm.forb", "p.cool.forb", "p.warm.forb")
+  Succulents <- c("succulents")
+  Grasses_C3 <- c("p.cool.grass")
+  Grasses_C4 <- c("p.warm.grass")
+  Grasses_Annuals <- c("a.cool.grass", "a.warm.grass")
+  Trees <- c()
+
+  # move to dist folder so we can begin adjusting space
+  setwd("../STEPWAT_DIST")
+
+  # will store the new rgroup files temporarily
+  new_rgroup_files <- c()
+
+  # Loop through all of the rgroup files defined in inputs
+  for(rg in rgroups){
+    #read the start of the file (where space is defined)
+    rgrp <- readLines(con <- paste0(rg,".in"), n = 10)
```

Comment:
might there be a way to not hard-code 10 resource groups here?

Reply:
Commit 9acfa2b allows for any number of RGroups to be specified in the input CSV. Our biggest limitation now is the template file, which prevents us from specifying different RGroups during the same run of the wrapper.

Reply:
@chaukap I was not requesting different RGroups during the same run of the wrapper, but potentially different RGroups (and RGroup counts) for different users who are simulating different systems.

Reply:
@kpalmqui In that case 9acfa2b does exactly what we need it to. The number of RGroups will be derived while reading the input CSV.